include/fiddle/postprocess/surface_values.h

```diff
@@ -0,0 +1,88 @@
+#ifndef included_fiddle_postprocess_surface_values_h
+#define included_fiddle_postprocess_surface_values_h
+
+#include <fiddle/base/config.h>
+
+FDL_DISABLE_EXTRA_DIAGNOSTICS
+#include <deal.II/base/mpi_remote_point_evaluation.h>
+FDL_ENABLE_EXTRA_DIAGNOSTICS
+#include <deal.II/base/point.h>
+#include <deal.II/base/smartpointer.h>
+#include <deal.II/base/tensor.h>
+
+#include <deal.II/dofs/dof_handler.h>
+
+#include <deal.II/fe/mapping.h>
+
+#include <vector>
+
+namespace fdl
+{
+  using namespace dealii;
+  using namespace SAMRAI;
+
+  /**
+   * Convenience class for interpolating values from the Cartesian grid onto a
+   * specified mesh.
+   */
+  template <int dim, int spacedim = dim>
+  class SurfaceValues
+  {
+  public:
+    /**
+     *
+     */
+    SurfaceValues(
+      const parallel::shared::Triangulation<dim, spacedim> &tria,
+      const Mapping<dim, spacedim>                         &mapping,
+      const Quadrature<dim>                                &quadrature,
+      tbox::Pointer<hier::BasePatchHierarchy<spacedim>>     patch_hierarchy,
+      const int                                             level_number);
+
+    /**
+     * Interpolate a scalar data index @p data_idx onto the finite element space
+     * and return the result.
+     */
+    LinearAlgebra::distributed::Vector<double>
+    interpolate_scalar(const int data_idx);
+
+    /**
+     * Interpolate a vector data index @p data_idx onto the finite element space
+     * and return the result.
+     */
+    LinearAlgebra::distributed::Vector<double>
+    interpolate_vector(const int data_idx);
+
+  protected:
+    /**
+     * Pointer to the provided triangulation.
+     */
+    SmartPointer<const parallel::shared::Triangulation<dim, spacedim>> tria;
+
+    /**
+     * Pointer to the provided mapping.
+     */
+    SmartPointer<const Mapping<dim, spacedim>> mapping;
+
+    Quadrature<dim> quadrature;
+
+    tbox::Pointer<hier::BasePatchHierarchy<spacedim>> patch_hierarchy;
+
+    int level_number;
+
+    std::unique_ptr<FiniteElement<dim, spacedim>> scalar_fe;
+
+    std::unique_ptr<FiniteElement<dim, spacedim>> vector_fe;
+
+    std::unique_ptr<DoFHandler<dim, spacedim>> scalar_dof_handler;
+
+    std::unique_ptr<DoFHandler<dim, spacedim>> vector_dof_handler;
```

Comment:
If we need to interpolate onto the mesh we probably need these so we can do quadrature, viz output, etc.